<a href="https://colab.research.google.com/github/oceanshrimp/py07/blob/master/notebook/mid_exam_py07.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 파이썬활용 중간고사 풀이

### 문제1. 다음 식의 연산 결과는?

In [1]:
a = 5
b = 3
a**b

125

#### a**b에서 **는 a의 b승을 말하므로 5의 3승은 125입니다.

### 문제2. 다음 문자열 a에 대하여 슬라이싱을 한 결과는?
           a = "Life is too short,you need Python!"

In [3]:
a = "Life is too short,you need Python!"
a[-16:-8]

'you need'

#### 슬라이싱에서 -1은 제일 뒤에 있는 숫자이다. 그리고 -1부터 차례대로보면      -16은 y, -8은 need 뒤 공백이다.하지만 포함을 안하므로 you need가 나온다.

### 문제3. 다음 list a의 인덱싱 결과는? 

In [4]:
a = [1,2,(3,4),['a','b',['My','Life','Good']]]
a[3][-1][0]

'My'

#### a[3]은 ['a','b',['My','Life','Good']]부분이고 여기서 -1은 ['My','Life','Good']이다. ['My','Life','Good']에서 0부분은 'My'이다. 

### 문제4.다음 tuple t1의 슬라이싱 결과는?

In [5]:
t1 = (0,1,"s", "t", "u")
t1[2:-2]

('s',)

#### t1[2:-2]에서 2는 "s", -2는 "t"이다. 하지만 -2는 포함되지않으므로 "s"인데 튜플에서 하나만 나오면 콤마를 붙여야된다. 그러므로 ("s",)이다

### 문제5. 다음 bool 값 출력 중 정확한 것은?

In [6]:
bool([0,1,2]),bool([0]), bool((0)), bool((0,))

(True, True, False, True)

#### bool값에서도 튜플인 경우에 하나가 나오면 예외없이 콤마를 붙여줘야되기 때문에 False가 나온다 

### 문제6. 다음 조건문을 1줄 코드로 변경하시오.

In [0]:
score = 80
if score >= 60:
  message = "success"
else:
  message = "failure"

In [26]:
score = 80
message = "success" if score>=60 else "failure"
print(message)

success


#### 조건문을 1줄 코드로 바꾸기 위해 조건부 표현식을 사용하면됩니다.

### 문제7. 다음 코드의 실행 결과는?

In [11]:
games = ["A", "B", "C"]

for cnt in range(len(games)):
    print(cnt, games[cnt], end=',')

0 A,1 B,2 C,

#### range(len(games))는 games의 요소 개수를 돌려주는 것인데 안에 3개니 3이 됩니다. 그러므로 cnt에 0부터 2까지 대입됩니다.

### 문제8. 다음의 결과가 출력되지 않는 것은? 'I ate 10 apples. so I was sick for three days.' 

In [24]:
number = 10
day = "three"
print("I ate {0} apples. so I was sick for {1} days.".format(10,day))
print(f"I ate {number} apples. so I was sick for {day} days.")
print(f"I ate {number} apples. so I was sick for {'three'} days.")
print("I ate {0} apples. so I was sick for {1} days.".format(number=10,day))

SyntaxError: ignored

#### 이미 number가 키워드로 지정되었기 때문에 포지션 지정을 할 수 없습니다. 포지션 지정이 키워드 지정보다 먼저 나와야 합니다


### 문제9. 리스트 내포를 사용해 1부터 1000이하의 자연수 중 5의 배수의 합을 구하는 한 줄 코드를 작성하시오.

In [27]:
sum([num for num in range(1,1001) if num % 5 ==0])

100500

#### if num % 5 ==0으로 5의 배수를 구하고 range(1,1001)로 1부터 1000이하의 자연수를 나타내어 sum으로 합을 구하는 형태입니다.

### 문제10. 4개 과목의 평균 성적을 구하시오. [a],[b]에 적합한 코드는?

In [0]:
subjects = ['python','english','math','science']
scores = [88,77,66,95]
total= 0

for subject, score in [a]._______(subjects, scores):
    total += score
    print(subject, score)

print("평균 점수 = {0}".format([b].__________))

In [0]:
[a] = zip(subjects, scores) [b] = total/len(subjects)

#### zip함수는  zip(subjects, scores)나 list(zip(subjects, scores))같은 형태로 만들어집니다. 그리고 평균점수는 문자열포매팅에 전체 합에 subjects의 요소 개수를 나눠야 됩니다.